microcosm_flask/conventions/catchall.py

```diff
@@ -0,0 +1,26 @@
+"""
+Flask's default behavior is to return a 404 for a route the app can't match.
+For backend services that are not queried directly by a front-end it can be useful
+to specify a different HTTP code, to differentiate between a nonexistent resource
+on a well-understood path (e.g. GET /user/123) and a nonexistent (e.g. GET /ursr).
+
+In particular, this can allow a reverse-proxy to identify cases where it is routing a request
+based on an outdated DNS entry.
+
+
+"""
+from microcosm.api import binding, defaults, typed
+
+
+@binding("catchall_convention")
+@defaults(
+    fallback_http_code=typed(int, default_value=421),
+)
+def configure_catchall_convention(graph):
+    fallback_http_code = graph.config.catchall_convention.fallback_http_code
+
+    @graph.flask.route("/<path:path>", methods=["GET", "POST", "PATCH", "PUT", "DELETE"])
+    def catch_all(path):
+        return "Unknown path", fallback_http_code
```

Comment:
Will this be logged in the same way as other requests?

Reply:
After double-checking: I believe so – the audit logging setup is for all Flask routes.

Reply:
Now it will be – I added the audit decorator on this route.

Reply:
@afallou few things:

1. can the audit include the unhandled route?
2. not sure what `success` means here. Seems like it should be `false` in this particular case
3. tiny nit: can we adjust the format of `elapased_time`? Bring it more up to date like other elapsed time field where the default unit of measurement is `ms` instead of `s`.

Reply:
@mlcohen I changed the logic to raise an `HTTPException`, that way the audit log will return `success: false` and include the response body (I added the path in it).
As for 3. the `elapsed_time` is already in ms, it's just fast 😄 .